cocotb/clock.py

```diff
@@ -51,16 +51,18 @@ def log(self):
 class Clock(BaseClock):
     r"""Simple 50:50 duty cycle clock driver.
 
-    Instances of this class should call its :meth:`start` method and :func:`fork` the
-    result.  This will create a clocking thread that drives the signal at the
+    Instances of this class should call its :meth:`start` method
+    and pass the coroutine object to one of the functions in :ref:`task-management`.
+
+    This will create a clocking task that drives the signal at the
     desired period/frequency.
 
     Example:
 
     .. code-block:: python
 
         c = Clock(dut.clk, 10, 'ns')
-        cocotb.fork(c.start())
+        cocotb.start_soon(c.start())
```

Comment:
You use `start_soon` here, but `start` in the other example. The task doesn't actually set an initial value on the clock until the body runs, does it? So if you use `start_soon` and then try to access the signal afterwards it may still be uninitialized. Of course the real problem is that the initialization doesn't happen in the `__init__` because `start_high` is an argument on the `start` method instead of an argument to the driver... Accepting that, we should recommend using `start` here.